### DISCUSSION TITLE
Firmware source code?

### DISCUSSION THREAD

**Discussion by oblique** [Anime4000/RTL960x#129] [General]
Hello,

I bought ODI DFP-34X-2C2 and I was wondering if anyone has the sources of the firmware, or if there are any open source firmware.

In general I'm not sure how much I should trust the device.

**Comment by Anime4000** [comment]
Hi, Realtek not release their firmware source code, only thing we can do is Reverse Engineering

**Comment by Ulkizz** [comment]
well... actually there were some sources leak some time ago. Unfortunately realtek banned those.

**Reply by Anime4000** [reply]
also no way to compile it, normal I have tried normal MIPS toolchain just simple hellow world, but wont run
